landside/catkin_ws/src/gui/src/gui/launch_dialog.py

```diff
@@ -87,12 +92,22 @@ def setup_node_name_box(self, selected_package):
         launchable_names = [os.path.split(f)[1] for f in launchables]
         self.node_name_box.addItems(launchable_names)
 
-    def node_name_selected(self, item_index):
+        try:
+            index_of_first_script_file = next(i for i, v in enumerate(launchable_names) if v.endswith('.py'))
```

Comment:
I'm struggling with order of operations here, could we get some parentheses. Also, you seem to be using exceptions for control flow. That's generally a bad idea.

Reply:
I removed the try/except that you highlighted. I also added parentheses to the if statement on line 97. Were you suggesting to add parentheses somewhere else?

There are two other places where I use exceptions for control flow: line 138 (parsing JSON) and line 214 (validating regex). I removed the try/except on line 214 and replaced it with an if statement validating the regex.

Using a try/except to ensure JSON is valid seems to be the most accepted (maybe only) way to validate JSON in python [(see this StackOverflow post)](https://stackoverflow.com/questions/5508509/how-do-i-check-if-a-string-is-valid-json-in-python).